### PR TITLE
add Create Task Template

### LINKED ISSUE #201: `CreateTaskTemplate`
The goal of this issue is to create a new gRPC service called `task-template-svc` with the corresponding Protobuf for the `CreateTaskTemplate` method inside the `task-svc` microservice.

The model of a TaskTemplate is already defined in https://github.com/helpwave/services/issues/200. 

Here is an outline for the Protobuf
```protobuf
message CreateTaskRequest {
  string name = 1;
  string description = 2;
  bool public = 3;
  optional string ward_id = 4;
  optional SubTask subtasks = 5; // When passed, every subtask is a seperate entity

  message SubTask {
    string name = 1;
  }
}
```

### PR DESCRIPTION
**Pull request by DasProffi**
## Which issues does this pull request close?
closes #201

### CODE REVIEW

**Comment by DasProffi** [comment]
The method fail because of the ward foreign key constraint being nullable, how do we fix this @MaxSchaefer or do we have solution already somewhere?

**Comment by DasProffi** [comment]
@MaxSchaefer the proto bufs fail although nothing changed

**Comment by use-to** [comment]
WTF is that commit message?!

**Comment by MaxSchaefer** [comment]
@DasProffi I've checked that ward_id Issue, was not aware that you already fixed that. Take a look https://github.com/helpwave/services/pull/202/commits/fada8ea152374746ce686e89cd2c7f5e8626d788

**Comment by DasProffi** [comment]
> @DasProffi I've checked that ward_id Issue, was not aware that you already fixed that. Take a look [fada8ea](https://github.com/helpwave/services/commit/fada8ea152374746ce686e89cd2c7f5e8626d788)

Thank you, that avoids the code duplication

File: services/task-svc/internal/task-template/task_template.go [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+		return nil, err
+	}
+
+	wardID, err := hwutil.StringToUUIDPtr(req.WardId)
+	if err != nil {
+		return nil, err
+	}
+
+	// authenticate the userID
+
```
Comment (use-to) [inline services/task-svc/internal/task-template/task_template.go:58]:
```suggestion
```

File: services/task-svc/migrations/000010_init.up.sql [hunk 2]
```diff
@@ -0,0 +1,20 @@
+CREATE TABLE IF NOT EXISTS task_templates (
+	id uuid PRIMARY KEY DEFAULT uuid_generate_v4(),
+	name text NOT NULL,
+	description text  NOT NULL,
+	is_public boolean NOT NULL,
+    ward_id uuid DEFAULT NULL,
+    user_id uuid NOT NULL,
```
Comment (use-to) [inline services/task-svc/migrations/000010_init.up.sql:7]:
```suggestion
	ward_id uuid DEFAULT NULL,
	user_id uuid NOT NULL,
```
indentation

**Review by use-to — CHANGES_REQUESTED** [review]

**Review by MaxSchaefer — CHANGES_REQUESTED** [review]

**Review by MaxSchaefer — APPROVED** [review]
Nice. When this is merged into main, I will work on that in https://github.com/helpwave/services/pull/205 to adopt the new ward service.

**Pull request merged**